src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -49,11 +49,11 @@ public class AsyncCursor<TDocument> : IAsyncCursor<TDocument>, ICursorBatchInfo
         private readonly int? _batchSize;
         private readonly CollectionNamespace _collectionNamespace;
         private IChannelSource _channelSource;
-        private bool _closed;
+        private volatile bool _closed;
```

Comment:
As I see this (and below) change is not required to make this ticket test pass. However, if I'm not mistaken, there is no harm to add it too.

Reply:
Hm this is a good question.  Eric Lippert apparently isn't a fan of volatile: https://stackoverflow.com/questions/72275/when-should-the-volatile-keyword-be-used-in-c
And some people suggest adding a lock around the variables as a safer alternative:
https://stackoverflow.com/questions/8927878/what-is-the-correct-way-of-adding-thread-safety-to-an-idisposable-object
I feel like using a lock is probably safer, but I'll defer to @rstam on this.

Reply:
If we're worried about a potential race condition, we could try running a test application that runs `Cursor_should_call_close_in_thread_safe_way` some high number of times.

Reply:
I disagree that it causes no harm to add volatile.

Adding volatile implies that you have a reason to add it, something presumably related to thread safety.

AsyncCursor is NOT thread safe. It is not intended to be thread safe and we won't be making any changes to make it thread safe.